src/ComponentBase.ts

```diff
@@ -396,18 +378,44 @@ export abstract class ComponentBase<P extends {}, S extends _.Dictionary<any>> e
         return false;
     }
 
+    private _findMatchingSubscription(subscriptions: Dictionary<StoreSubscriptionInternal<P, S>>): boolean {
```

Comment:
This method is only used in one place. It seems a bit premature to extract out of its context. For example, the name is confusing. You say "find" but it returns a `boolean`; should it be "has" instead? I am still not sure what "matching" means. More evidence that this logic is not ready to be its own top-level method out of context.

Reply:
Removed `_findMatchingSubscription`